zap/src/config.rs

```diff
@@ -482,7 +487,13 @@ impl NumTy {
 			NumTy::U32
 		} else {
 			NumTy::F64
-		}
+		};
+
+		(numty, offset)
```

Comment:
It's weird to return a tuple from `from_f64` and then have to invent a new method `from_f64_max` to *not* do that.

Consider either:
1. NumTy becomes a `struct` with an optional range and then the type of number
2. NumTy stays as an enum, but inside the args of each item it has an optional range